demos/projects/ESPRESSIF/aziotkit/main/azure_iot_freertos_esp32_sensors.c

```diff
@@ -0,0 +1,390 @@
+    configASSERT( xAzIoTResult == eAzureIoTSuccess );
+
+    xAzIoTResult = AzureIoTJSONWriter_AppendPropertyWithDoubleValue( &xWriter, ( uint8_t * ) sampleazureiotgsgTOTAL_MEMORY_PROPERTY_NAME, sizeof( sampleazureiotgsgTOTAL_MEMORY_PROPERTY_NAME ) - 1,
+                                                                     xTotalMemoryPropertyValue, 0 );
+    configASSERT( xAzIoTResult == eAzureIoTSuccess );
+
+    xAzIoTResult = AzureIoTHubClientProperties_BuilderEndComponent( &xAzureIoTHubClient, &xWriter );
+    configASSERT( xAzIoTResult == eAzureIoTSuccess );
+
+    xAzIoTResult = AzureIoTJSONWriter_AppendEndObject( &xWriter );
+    configASSERT( xAzIoTResult == eAzureIoTSuccess );
+
+    lBytesWritten = AzureIoTJSONWriter_GetBytesUsed( &xWriter );
+
+    if( lBytesWritten < 0 )
+    {
+        LogError( ( "Error getting the bytes written for the device properties JSON" ) );
+        lBytesWritten = 0;
+    }
+
+    return lBytesWritten;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Command message callback handler
+ */
+uint32_t ulSampleHandleCommand( AzureIoTHubClientCommandRequest_t * pxMessage,
+                                uint32_t * pulResponseStatus,
+                                uint8_t * pucCommandResponsePayloadBuffer,
+                                uint32_t ulCommandResponsePayloadBufferSize)
+{
+    uint32_t ulCommandResponsePayloadLength;
+
+    ESP_LOGI( TAG, "Command payload : %.*s \r\n",
+              pxMessage->ulPayloadLength,
+              ( const char * ) pxMessage->pvMessagePayload );
+
+    if ( strncmp( ( const char * ) pxMessage->pucCommandName, sampleazureiotCOMMAND_TOGGLE_LED1, pxMessage->usCommandNameLength ) == 0)
+    {
+        xLed1State = !xLed1State;
+        toggle_wifi_led( xLed1State ? LED_ON : LED_OFF );
+
+        *pulResponseStatus = AZ_IOT_STATUS_OK;
+        ulCommandResponsePayloadLength = sizeof( sampleazureiotCOMMAND_EMPTY_PAYLOAD ) - 1;
+        (void)memcpy( pucCommandResponsePayloadBuffer, sampleazureiotCOMMAND_EMPTY_PAYLOAD, ulCommandResponsePayloadLength );
+    }
+    else if ( strncmp( ( const char * ) pxMessage->pucCommandName, sampleazureiotCOMMAND_TOGGLE_LED2, pxMessage->usCommandNameLength ) == 0)
+    {
+        xLed2State = !xLed2State;
+        toggle_azure_led( xLed2State ? LED_ON : LED_OFF );
+
+        *pulResponseStatus = AZ_IOT_STATUS_OK;
+        ulCommandResponsePayloadLength = sizeof( sampleazureiotCOMMAND_EMPTY_PAYLOAD ) - 1;
+        (void)memcpy( pucCommandResponsePayloadBuffer, sampleazureiotCOMMAND_EMPTY_PAYLOAD, ulCommandResponsePayloadLength );
+    }
+    else if ( strncmp( ( const char * ) pxMessage->pucCommandName, sampleazureiotCOMMAND_DISPLAY_TEXT, pxMessage->usCommandNameLength ) == 0)
+    {
+        oled_clean_screen();
+        oled_show_message( ( const uint8_t * ) pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );

```

Comment:
drop the first and last character in the payload as this contains the " characters which we dont want printed to the screen.